qutip/solver/parallel.py

```diff
@@ -9,13 +9,18 @@
 import sys
 import time
 import threading
+import concurrent.futures
 from qutip.ui.progressbar import progess_bars
 from qutip.settings import available_cpu_count
 
 if sys.platform == 'darwin':
-    Pool = multiprocessing.get_context('fork').Pool
+    mp_context = multiprocessing.get_context('fork')
```

Comment:
For mac, we used `fork` since `spawn` would fail pickling, but it may work with `concurrent`. We should try it eventually.

Reply:
I *think* spawn would actually work (or at least it works for our Windows tests). Either `spawn` or `forkserver` do require more complex data sharing since things are no longer in the same process. I think moving data around is what kills performance (Linux correlation tests timeout with `forkserver` too).